connection/cluster/ClusterClient.java

```diff
@@ -93,13 +93,14 @@ private Map<String, ClusterServerClient> createClients(TypeDBCredential credenti
         Map<String, ClusterServerClient> clients = new HashMap<>();
         boolean available = false;
         for (String address : addresses) {
+            ClusterServerClient client = new ClusterServerClient(address, credential, parallelisation);
             try {
-                ClusterServerClient client = new ClusterServerClient(address, credential, parallelisation);
-                clients.put(address, client);
+                client.open();
```

Comment:
this 'attempt's to open the connection, but we retain the client afterward anyway